digitalocean/datasource_digitalocean_database_replica.go

```diff
@@ -0,0 +1,110 @@
+package digitalocean
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func dataSourceDigitalOceanDatabaseReplica() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceDigitalOceanDatabaseReplicaRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+
+			"cluster_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"size": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"private_network_uuid": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				Computed:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+
+			"host": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"private_host": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"port": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"uri": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+
+			"private_uri": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+
+			"database": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"user": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"password": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+
+			"tags": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validateTag,
+				},
+				Set: HashStringIgnoreCase,
+			},
+		},
+	}
+}
+
+func dataSourceDigitalOceanDatabaseReplicaRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	d.Set("cluster_id", d.Get("cluster_id").(string))
+	d.Set("name", d.Get("name").(string))
```

Comment:
```suggestion
	clusterId :=  d.Get("cluster_id").(string)
	name := d.Get("name").(string)
	d.Set("cluster_id", clusterId)
	d.Set("name", name)
	d.SetId(makeReplicaId(clusterId, name))
```

You'll need to set a unique ID for the data source. We already have a little helper to make one from the cluster ID and replica name:

https://github.com/digitalocean/terraform-provider-digitalocean/blob/c73e7a5cf4fbe88060c471a808acf862c327b893/digitalocean/resource_digitalocean_database_replica.go#L208-L210

Reply:
Ah nice.

Reply:
thanks andrew!